stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -2368,6 +2376,10 @@ internal constructor(
             userCredentialStorage = credentialStorage
         }
 
+        public fun withApiRequestAnalyser(apiRequestsAnalyser: ApiRequestsAnalyser): Builder = apply {
```

Comment:
What do you think about adding a parameter to check if this only works in Debug mode like below?

```kotlin
public fun withApiRequestAnalyser(apiRequestsAnalyser: ApiRequestsAnalyser, debugOnly: Boolean = true): Builder = apply {
     if (debugOnly && BuildConfig.DEBUG) {
         this.apiRequestsAnalyser = apiRequestsAnalyser
     }
}
```

We need to prevent being enabled this option by mistake and clients don't need to split up their initialization codes to check whether it is a debug mode.